File: src/Pirate.Lexer.TokenType/Enums/TokenGroup.cs
using System.Text.Json.Serialization;
using Newtonsoft.Json.Converters;

namespace Pirate.Lexer.TokenType.Enums;

/// <summary>
/// A enum for token groups.
/// </summary>
[JsonConverter(typeof(StringEnumConverter))]
public enum TokenGroup
{
    OPERATORS,
    COMPARISONOPERATORS,
    TYPEKEYWORD,
    CONTROLKEYWORD,
    SYNTAX,
    VALUE,
    Empty
}

File: src/Pirate.Lexer.Enums/TokenGroup.fs
namespace Pirate.Lexer.Enums

open Newtonsoft.Json
open Newtonsoft.Json.Converters

[<JsonConverter(typeof<StringEnumConverter>)>]
type TokenGroup =
    | OPERATORS
    | COMPARISONOPERATORS
    | TYPEKEYWORD
    | CONTROLKEYWORD
    | SYNTAX
    | VALUE
    | Empty